inspirehep/modules/orcid/converter.py

```diff
@@ -0,0 +1,245 @@
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""
+Handle conversion from INSPIRE records to ORCID.
+"""
+
+from __future__ import absolute_import, division, print_function
+from inspire_utils.record import get_value
+from inspire_utils.date import PartialDate
+from .builder import OrcidBuilder
+from flask import current_app
+import re
+from inspirehep.modules.hal.utils import (
+    get_journal_title,
+    get_publication_date,
+    get_conference_title,
+    get_conference_record,
+    get_conference_country,
+    get_doi,
+)
+
+
+class OrcidConverter(object):
+    """Coverter for the Orcid format."""
+
+    # Maps INSPIRE author roles to ORCID contributor roles
+    # for full list see: https://git.io/vdKXv#L235-L245
+    INSPIRE_TO_ORCID_ROLES_MAP = {
+        'author': 'author',
+        'editor': 'editor',
+        'supervisor': None
+    }
+
+    # Maps INSPIRE document type to ORCID work types
+    # for full list see: https://git.io/vdKXv#L118-L155
+    INSPIRE_DOCTYPE_TO_ORCID_TYPE = {
+        'article': 'journal-article',
+        'book': 'book',
+        'book chapter': 'book-chapter',
+        'conference paper': 'conference-paper',
+        'note': 'other',
+        'proceedings': 'edited-book',
+        'report': 'report',
+        'thesis': 'dissertation',
+    }
+
+    def __init__(self, record, put_code=None, visibility=None):
+        """Constructor.
+
+        Args:
+            record (dict): a record.
+        """
+        self.record = record
+        self.put_code = put_code
+        self.visibility = visibility
+
+    def get_xml(self):
+        """Create an ORCID XML representation of the record.
+
+        Returns:
+            lxml.etree._Element: ORCID XML work record
+        """
+        builder = OrcidBuilder()
+
+        # Set attributes
+        if self.visibility:
+            builder.set_visibility(self.visibility)
+
+        if self.put_code:
+            builder.set_put_code(self.put_code)
+
+        # Add a title
+        if self.title:
+            builder.add_title(self.title, self.subtitle, self.title_translation)
+
+        # Add a journal title
+        containing_publication_title = self.journal_title or self.conference_title or self.book_series_title
+        if containing_publication_title:
+            builder.add_journal_title(containing_publication_title)
+
+        # TODO: Add a citation
```

Comment:
What's this? Is there somewhere where can I read what was supposed to be here?

Reply:
I wrote this before we had the BibTex sorted, so now you actually just pass a citation (i.e. BibTex) string to `builder.add_citation`.

Reply:
:+1: